packages/bpk-component-card-list/src/BpkCardList.tsx

```diff
@@ -0,0 +1,165 @@
+import { BpkAccessoryTypes, type BpkCardListProps } from './common-types';
+
+import STYLES from './BpkCardList.module.scss';
+
+const getClassName = cssModules(STYLES);
+const MAX_ITEMS = 12; // MAX should be 12 for Desktop Grid and Mobile Stack
+const DEFAULT_ITEMS = 3;
+
+const BpkCardList = (props: BpkCardListProps) => {
+  const {
+    buttonText,
+    cardList,
+    description,
+    initiallyShownCards = DEFAULT_ITEMS,
+    layoutDesktop,
+    layoutMobile,
+    onButtonClick,
+    title,
+  } = props;
+
+  const [visibleCards, setVisibleCards] = useState(
+    cardList.slice(0, initiallyShownCards),
+  );
+  const [collapsed, setCollapsed] = useState(true);
+
+  const showContent = () => {
+    setVisibleCards(cardList.slice(0, MAX_ITEMS));
+
+    onButtonClick?.();
+  };
+
+  const hideContent = () => {
+    setVisibleCards(cardList.slice(0, initiallyShownCards));
+
+    onButtonClick?.();
+  };
+
+  const button = buttonText && (
+    <BpkButtonV2 onClick={onButtonClick}>{buttonText}</BpkButtonV2>
+  );
+
+  return (
+    <div className={getClassName('bpk-card-list')}>
+      <BpkSectionHeader
+        title={title}
+        description={description}
+        button={button}
+      />
+
+      <div className={getClassName('bpk-card-list--card-list')}>
+        <BpkBreakpoint query={BREAKPOINTS.MOBILE}>
+          {(isActive) => {
+            if (isActive) {
+              if (layoutMobile === 'rail') {
+                return <BpkCardListRail>{cardList}</BpkCardListRail>;
+              }
+              const { accessory } = props;
+
+              let modeProps = {};
+              if (accessory === BpkAccessoryTypes.Expand) {
+                modeProps = {
+                  accessory,
+                  expandText: props.expandText,
+                  onButtonClick: props.onButtonClick,
+                };
+              } else if (accessory === BpkAccessoryTypes.Button) {
+                modeProps = {
+                  accessory,
+                  buttonText: props.buttonText,
+                  onButtonClick: props.onButtonClick,
+                };
+              }
+              return (
+                <BpkCardListStack
+                  showContent={showContent}
+                  hideContent={hideContent}
+                  collapsed={collapsed}
+                  setCollapsed={setCollapsed}
+                  {...modeProps}
+                >
+                  {visibleCards}
+                </BpkCardListStack>
+              );
+            }
+
+            if (layoutDesktop === 'row') {
+              const { accessory } = props;
+              return (
+                <BpkCardListRow
+                  accessory={button ? undefined : accessory}
+                  numberOfCardsToShow={initiallyShownCards}
+                  ariaLabelNext={props.ariaLabelNext}
+                  ariaLabelPrev={props.ariaLabelPrev}
+                  ariaLabelIndicator={props.ariaLabelIndicator}
```

Comment:
Same here

Reply:
This one its probably better to deconstruct with accessory. Will do that now